specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json

```diff
@@ -0,0 +1,5736 @@
+          "type": "string"
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "Properties of the Storage account data set.",
+      "required": [
+        "storageAccountResourceId",
+        "subscriptionId",
+        "resourceGroup",
+        "storageAccountName"
+      ],
+      "type": "object",
+      "properties": {
+        "dataSetId": {
+          "description": "Unique id for identifying a data set resource",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Location of the storage account.",
+          "type": "string"
+        },
+        "paths": {
+          "description": "A list of paths under a given storage account.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageAccountPath"
+          }
+        },
+        "resourceGroup": {
+          "description": "Resource group of storage account",
+          "type": "string"
+        },
+        "storageAccountName": {
+          "description": "Storage account name of the source data set",
+          "type": "string"
+        },
+        "storageAccountResourceId": {
+          "description": "Resource id of the storage account.",
+          "type": "string"
+        },
+        "subscriptionId": {
+          "description": "Subscription id of storage account",
+          "type": "string"
+        }
+      }
+    },
+    "SynapseSparkPipelineStep": {
+      "description": "A Synapse Spark based pipeline step.",
+      "required": [
+        "properties",
+        "kind"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/PipelineStep"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SynapseSparkPipelineStepProperties",
+          "description": "Properties of Synapse Spark pipelineStep",
+          "x-ms-client-flatten": true
+        }
+      },
+      "x-ms-discriminator-value": "SynapseSparkPipelineStep"
+    },
+    "SynapseSparkPipelineStepProperties": {
+      "description": "A Synapse Spark PipelineStep data transfer object.",
+      "required": [
+        "clusterSize",
+        "scriptId",
+        "scriptRevision",
+        "scriptSourceBindings",
+        "scriptSinkBindings"
+      ],
+      "type": "object",
+      "properties": {
+        "clusterSize": {
+          "format": "int32",
+          "description": "Defines the cluster size for running a SynapseSpark pipeline step",
+          "type": "integer"
+        },
+        "dependencies": {
+          "description": "List of other steps names, in the same pipeline, which this step is depended on\r\nand should run only when they finish.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "scriptId": {
```

Comment:
why is this using the generated GUIDs for referencing instead of resource IDs?  It means downstream systems won't be able to generically follow these references

Reply:
resource Ids are less strict IIUC. A script can be deleted and another can be created with the same name, we also 
support move operation automatically, no need to update every reference with the new id.

Another thing is - this can be a script from another tenants subscritpion where the user don't even have access to, they get access via the entitltment.